Backend/Controllers/Controller.cs

```diff
@@ -281,8 +336,195 @@ public IActionResult RunAlgorithm(AlgorithmRunParameters algorithmRunParameters)
+            {
+                string json = System.IO.File.ReadAllText(lastQueryPath);
+                algorithmRunParameters = JsonConvert.DeserializeObject<AlgorithmRunParameters>(json);
+            }
+            else
+            {
+                return NotFound();
+            }
+            
+            string reportsFolder = System.IO.Path.Combine(Directory.GetCurrentDirectory(), "Reports");
+
+            // usuwanie folderu z raportami żeby nie zwracało poprzednich
+            //if (Directory.Exists(reportsFolder))
+            //{
+            //    Directory.Delete(reportsFolder, true);
+            //}
+
+            string[] optimizationAlgorithmNames = algorithmRunParameters.OptimizationAlgorithmNames;
+            string[] testFunctionNames = algorithmRunParameters.TestFunctionNames;
+            int dim = algorithmRunParameters.Dim;
+            List<ParamForAlgorithm> paramsForAlgorithm = algorithmRunParameters.paramsForAlgorithm;
+
+            string testFunctionsFolder = System.IO.Path.Combine(Directory.GetCurrentDirectory(), "TestFunctions");
+            string optimizationAlgorithmsFolder = System.IO.Path.Combine(Directory.GetCurrentDirectory(), "OptimizationAlgorithms");
+            
+            if (testFunctionNames.Length == 1 && optimizationAlgorithmNames.Length != 1)
+            {
+                int iAStart = 0;
+                
+                string testStatePath = System.IO.Path.Combine(stateFolder, "TestState.json");
+                TestState testState = new TestState();
+
+                if (System.IO.File.Exists(testStatePath))
+                {
+                    string json2 = System.IO.File.ReadAllText(testStatePath);
+                    testState = JsonConvert.DeserializeObject<TestState>(json2);
+                    iAStart = testState.AlgorithmIterator;
+                }
+
+                string testFunctionDLL = SearchDLLs.SearchDLLsInDirectory(testFunctionNames, testFunctionsFolder)[0];
+                string[] optimizationAlgorithmDLLs = SearchDLLs.SearchDLLsInDirectory(optimizationAlgorithmNames, optimizationAlgorithmsFolder);
+
+                //testing started
+                isFinished = false;
+
+                using (StreamWriter writer = new StreamWriter(isFinishedPath))
+                {
+                    writer.WriteLine(isFinished);
+                }
+
+                for (int iA = iAStart; iA < optimizationAlgorithmDLLs.Length; iA++)
+                {
+                    var oneOptimizationAlgorithmDLL = optimizationAlgorithmDLLs[iA];
+                    Backend.RunAlgorithm.Run(oneOptimizationAlgorithmDLL, new string[] { testFunctionDLL }, dim, paramsForAlgorithm);
+
+                    testState.AlgorithmIterator = iA + 1;
+                    testState.TestFuncIterator = 0;
+                    testState.ParamIterator = 0;
+                    testState.Iterator = 0;
+                    testState.BestData = null;
```

Comment:
To na pewno było wyżej. Popatrz czy nie da się powtórzonego kodu wrzucić do funkcji